CMake Step fails if build directory is outside of the clone

As found by @kgerheiser and @aoloso, if your `build/` directory isn't in the clone, CMake fails. The fix is simple.

The error is:
```
fatal: not a git repository (or any parent up to mount point /gpfsm)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
CMake Error at src/Applications/GEOSdas_App/testsuites/CMakeLists.txt:33 (message):
  This should not be reached

```

--- a/src/Applications/GEOSdas_App/testsuites/CMakeLists.txt
+++ b/src/Applications/GEOSdas_App/testsuites/CMakeLists.txt
@@ -20,6 +20,7 @@ if(GIT_FOUND)
       execute_process(
          COMMAND ${GIT_EXECUTABLE} merge-base --is-ancestor ${Project_WC_LATEST_TAG} ${Project_WC_REVISION_HASH}
          RESULT_VARIABLE is_latest_tag_an_ancestor
+         WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
          )
       if (is_latest_tag_an_ancestor EQUAL 0)
          #message("Tag ${Project_WC_LATEST_TAG} is an ancestor of ${Project_WC_REVISION_HASH}")

